subprojects/core/src/main/java/org/gradle/api/internal/provider/DefaultProviderFactory.java

```diff
@@ -69,9 +67,9 @@ public DefaultProviderFactory(FileOperations fileOperations) {
             ((PropertyState<Double>) propertyState).set(Double.valueOf(0));
         } else if (clazz == Character.class) {
             ((PropertyState<Character>) propertyState).set(new Character('\0'));
-        } else if (clazz == FileCollection.class || clazz == ConfigurableFileCollection.class) {
+        } else if (clazz == ConfigurableFileCollection.class) {
```

Comment:
I thought we had agreed on deferring this work to a later story. I added it to the spec here: https://github.com/gradle/gradle/blob/master/design-docs/internal-apis-made-public.md#story-filecollection-is-a-provider. `PropertyState<ConfigurableFileCollection>` is also used heavily in `PropertyStateIntegrationTest`. What would be the alternative?